scripts/intfutil

```diff
@@ -94,16 +94,21 @@ def get_interface_vlan_dict(config_db):
     return int_to_vlan_dict
 
 
-def config_db_vlan_port_keys_get(int_to_vlan_dict, front_panel_ports_list, intf_name):
+def config_db_vlan_port_keys_get(int_to_vlan_dict, front_panel_ports_list,config_db, intf_name):
     """
     Get interface vlan value and return it.
     """
     vlan = "routed"
+    ports_dict = config_db.get_table('PORT')
     if intf_name in front_panel_ports_list:
         if intf_name in int_to_vlan_dict.keys():
             vlan = int_to_vlan_dict[intf_name]
-            if "Vlan" in vlan:
-                vlan = "trunk"
+            if "PortChannel" in vlan:
+                return vlan
+    if intf_name in ports_dict:
+        intf_data = ports_dict[intf_name]
+        if "mode" in intf_data:
```

Comment:
We dont need 'mode' field, we can easily infer 'access'/'trunk' for a given port using VLAN_MEMBER table.

Reply:
If we do not use “mode attribute” and use “VLAN_MEMBER table” instead to implement the restriction of having no tagged members in access mode then it would need to parse the VLAN_MEMBER Table which can potentially have 10’s to 1000’s of entries whenever a vlan member is added. On the other hand, by using the mode attribute we can achieve the same result using a single get command without parsing the whole VLAN_MEMBER Table.  

This mode attribute will be useful to implement other port modes as well e.g. hybrid, which is part of 202305 feature release.

Reply:
I understand your point, let me think about the backward compatibility issues as well and update here.

Reply:
> We need the 'mode' field for portchannel as well not just for port.

So far we have focused on the physical ports only. The mode for portchannel can be added later in a separate PR.

---

> What should be the default 'mode' - 'none'?

Default mode is still “routed” as shown by the output of the “show interface status” command where it shows default behavior as routed.

---
> If we assign the IP on the port/portchannel, do we need to enforce the user to set 'routed' mode first?

In SONiC, IP assignment can be done only if the port has not assigned any VLAN. Earlier, we followed the same practice. But after your observation/concern on this, we have updated our code. We added a restriction on IP Assignment, the user needs to enforce the mode “routed” before IP assignment else it will not accept IP.

---
> Also, do we need to enforce mode config before associating the port/portchannel to VLAN(s)?

 Yes, we need to enforce mode config before associating the port to VLAN(s). For portchannel, the status is same as above answer for portchannel

---
> Do we need to restrict the mode deletion when the port is part of some VLANs?

Yes,  we need to restrict the mode deletion when a port is part of some VLAN(s)

---
> How do we handle backward compatibility? what if we load old configs after this change? are you planning to update db_migrator script?

We will get back to you about the db_migrator. Just to mention please note the following:
If we upgrade just the CLI on old versions of sonic and run commands like switch port mode, it will work and set the mode attribute in the PORT table.

---
> Please raise the SONiC YANG model pull request as well

Yes, we will raise a new PR for SONiC YANG Model.

Reply:
So far we have focused on the physical ports only. The mode for portchannel can be added later in a separate PR.
>> Please add the support for portchannel in this PR itself. This helps to perform the testing for all cases and we dont have to repeat the test cases again and will be easy for the user to migrate to the validation for VLAN members.

Reply:
Please add the support for portchannel in this PR itself. This helps to perform the testing for all cases and we dont have to repeat the test cases again and will be easy for the user to migrate to the validation for VLAN members.

> > We have updated our code along with HLD as per your suggestions, pls review for PortChannel.